src/cpp/src/model_runner.hpp

```diff
@@ -75,6 +79,11 @@ class ModelRunner {
         return m_request;
     }
 
+    void set_inputs_embedder(std::shared_ptr<InputsEmbedder> embedder) {
+        m_use_embeddings = true;
```

Comment:
looks like `m_use_embeddings`  is excessive as we can always check `if (m_inputs_embedder)`

Reply:
Done.